pkg/network/dns_stats.go

```diff
@@ -157,12 +163,20 @@ func (d *dnsStatKeeper) removeExpiredStates(earliestTs time.Time) {
 	defer d.mux.Unlock()
 	threshold := microSecs(earliestTs)
 	for k, v := range d.state {
-		if v < threshold {
+		if v.ts < threshold {
 			delete(d.state, k)
-			d.deleteCount++
-			stats := d.getStats(k.key)
+			allStats, ok := d.stats[k.key]
+			if !ok {
+				allStats = make(map[string]dnsStats)
+			}
+			stats, ok := allStats[v.question]
+			if !ok {
+				stats.countByRcode = make(map[uint8]uint32)
+				d.deleteCount++
```

Comment:
Previously this was incremented regardless of whether stats existed for `k.key` or not, why the change?

Reply:
The change was unintentional. Thanks for catching!